documentation/docs/extensibility.md

```diff
@@ -1,86 +1,214 @@
 # Extensibility
 
-There are several possibilities for extensibility besides the **[very powerful configuration](configuration.md)**:
+When using one of the ready-made pipelines of project "Piper", you don't have to write custom pipeline code.
+The pipelines are centrally maintained and can be used with only a small amount of declarative configuration as documented [here](configuration.md).
 
-## 1. Stage Exits
+For the vast majority of _standard_ projects, the features of the ready-made pipelines should be enough to implement [Continuous Delivery](https://martinfowler.com/bliki/ContinuousDelivery.html) with little effort in a best-practice compliant way.
+If you miss a feature or discover a bug in one of our pipelines, please see if there is already an [issue in our GitHub repository](https://github.com/SAP/jenkins-library/issues) and if not, open a new one.
 
-  You have to create a file like `<StageName>.groovy` (for example, `Acceptance.groovy`) and store it in folder `.pipeline/extensions/` in your source code repository.
+In some cases, it's not desirable to include specialized features in the ready-made pipelines.
+However, you can still benefit from their qualities, if you address your requirements through an **extension**.
+Extensions are custom bits of pipeline coding, which you can use to implement special requirements.
+Before building extensions, please make sure that there is no alternative that works better for you.
 
-!!! note "Cloud SDK Pipeline"
-    If you use the Cloud SDK Pipeline, the folder is named `pipeline/extensions/` (without the dot). For more information, please refer to [the Cloud SDK Pipeline documentation](https://github.com/SAP/cloud-s4-sdk-pipeline/blob/master/doc/pipeline/extensibility.md).
+Options for extensibility, in the order in which we recommend considering them:
 
-  The pipeline template checks if such a file exists and executes it, if present.
-  A parameter that contains the following keys is passed to the extension:
+## 1. Extend individual stages
 
-  * `script`: defines the global script environment of the Jenkinsfile run. This makes sure that the correct configuration environment can be passed to project "Piper" steps and also allows access to for example the `commonPipelineEnvironment`.
-  * `originalStage`: this will allow you to execute the "original" stage at any place in your script. If omitting a call to `originalStage()` only your code will be executed instead.
-  * `stageName`: name of the current stage
-  * `config`: configuration of the stage (including all defaults)
+In this option, you use the centrally maintained pipeline but can change individual stages, if required.
```

Comment:
```suggestion
In this option, you use the centrally maintained pipeline but can extend or even exchange individual stages, if required.
```

Reply:
I'm a bit hesitant to write this explicitly because it is more complicated to totally rewrite a stage. Users who are advanced enough to do that will understand it anyway, right?